fix(windows): install and load wintun.dll from a well-known path instead of setting the current directory

ReactorScram opened the pull request:
closes #3425

```[tasklist]
- [x] Switch to connlib-shared for BUNDLE_ID and stuff
- [x] Break out small things into other PRs if possible
- [x] Fix merge conflicts
```

ReactorScram committed:
refactor(windows): set absolute paths for logs and wintun.dll

ReactorScram committed:
tests(windows): change the semantics of the debug wintun subcmd

ReactorScram committed:
fix(windows): install and load wintun.dll from a well-known path instead of setting the current directory, closes #3425

ReactorScram committed:
docs(windows): explain more about why this change is needed

ReactorScram committed:
fix(windows): hide Windows code from other platforms

ReactorScram committed:
fix(windows): put the crash handler Unix domain socket in a well-known path

ReactorScram committed:
fix: compile error on Ubuntu

ReactorScram committed:
fix: cross-platform stuff

ReactorScram committed:
docs(windows): typo

jamilbk commented on rust/windows-client/src-tauri/src/client/crash_handling.rs:
```diff
@@ -80,8 +86,10 @@ fn start_server_and_connect() -> anyhow::Result<(minidumper::Client, std::proces
         }
 
         server = Some(
+            //
```
Typo?

ReactorScram replied:
Yeah, removed that. I was probably about to write `// TODO: pass the socket name to the server so we don't have to generate it in two places` and then just implemented that and left the slashes.

jamilbk reviewed (commented):
LGTM will wait for @conectado's approval / comments on code organization.

ReactorScram committed:
Update rust/windows-client/src-tauri/src/client/debug_commands.rs

Co-authored-by: Gabi <gabrielalejandro7@gmail.com>
Signed-off-by: Reactor Scram <ReactorScram@users.noreply.github.com>

ReactorScram committed:
Update rust/windows-client/src-tauri/src/client/wintun_install.rs

Co-authored-by: Gabi <gabrielalejandro7@gmail.com>
Signed-off-by: Reactor Scram <ReactorScram@users.noreply.github.com>

ReactorScram committed:
fix(windows): typo

ReactorScram committed:
Merge branch 'reactorscram/fix-paths' into reactorscram/fix-wintun-dll

Signed-off-by: Reactor Scram <ReactorScram@users.noreply.github.com>

ReactorScram committed:
fix(windows)

ReactorScram committed:
Merge branch 'reactorscram/fix-paths' into reactorscram/fix-wintun-dll

Signed-off-by: Reactor Scram <ReactorScram@users.noreply.github.com>

ReactorScram committed:
Merge remote-tracking branch 'origin/main' into reactorscram/fix-wintun-dll

ReactorScram committed:
Merge branch 'main' into reactorscram/fix-wintun-dll

ReactorScram committed:
Merge branch 'main' into reactorscram/fix-wintun-dll

Signed-off-by: Reactor Scram <ReactorScram@users.noreply.github.com>

conectado commented on rust/windows-client/firezone-windows-common/README.md:
```diff
@@ -0,0 +1,17 @@
+firezone-windows-common
```
I'd prefer if we use `connlib-shared` rather than adding a new crate.

conectado replied:
we can add a windows specific module

ReactorScram replied:
Cool, I'll revise this PR

ReactorScram committed:
Merge branch 'main' into reactorscram/fix-wintun-dll

ReactorScram committed:
Merge branch 'main' into reactorscram/fix-wintun-dll

ReactorScram committed:
wip(windows)

ReactorScram committed:
refactor(windows): change some anyhow errors into thiserror errors

This is part of handling the WebView-not-installed error, #3451

ReactorScram committed:
fix(windows): show error dialog if you try to run the exe without WebView2 installed

ReactorScram committed:
docs(windows): <https://github.com/firezone/firezone/issues/3451>

ReactorScram committed:
Merge remote-tracking branch 'origin/main' into reactorscram/fix-wintun-dll

ReactorScram committed:
refactor(windows): move the common functions into connlib_shared::windows

ReactorScram committed:
fix(windows): disable window module for non-windows OSes

`known_folders` can compile on other OSes but nothing will work, since
it's built atop the Windows API. Luckily only the Windows client needs this,
and it is already disabled on other OSes.

ReactorScram committed:
Merge branch 'main' into reactorscram/fix-wintun-dll

ReactorScram committed:
Merge remote-tracking branch 'origin/main' into reactorscram/fix-webview2-crash

ReactorScram committed:
Merge branch 'reactorscram/fix-webview2-crash' into reactorscram/fix-wintun-dll

ReactorScram committed:
Merge remote-tracking branch 'origin/main' into reactorscram/fix-wintun-dll
